### PR TITLE
feat: Optimize vacuum drop table

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. use batch operation to list dropped table or db
2. add [limit n] in vacuum drop table option.
3. unit test of limit option

- Closes https://github.com/datafuselabs/databend/issues/13543

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13576)

### CODE REVIEW

**Comment by BohuTANG** [comment]
We could merge this PR into main after our MERGE INTO is ready to use, make less risky? cc @dantengsky

**Review by drmingdrmer — APPROVED** [review]
> Reviewed 4 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13576)* status: 4 of 10 files reviewed, all discussions resolved (waiting on @dantengsky)

**Review by dantengsky — APPROVED** [review]

**Pull request merged**
